### PR TITLE
Typescript the contents Manager.

### PR DESCRIPTION
**Pull request by Carreau**
I'm mostly doing that to get the .d.ts file from the contents manager and apply it in jupyter_drive to be sure I have the same API (spoiler it seem not to be the case, but things still works). o_0. 

Todo:

 -  find a way to get ts -> correct folder
    - `a/b/c.ts` to `a/b/c.js`
    - `x/y/z.ts` to `x/y/z.js`
 - find correct way to extend Errors.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,5 +21,17 @@
     "requirejs": "^2.1.17",
     "source-map": "^0.4.2",
     "through": "^2.3.7"
+    "del": "^0.1.3",
```
Comment (takluyver) [inline package.json]: Need to add a comma on the line above

Comment (Carreau) [inline package.json]: Bad merge/rebase of package.json, there are some duplicate deps in here too. Will fix.

**Comment by Carreau** [comment]
migrated session.js, point to the fact that the gulpfile does not ut the js files in the right place.

**Comment by Carreau** [comment]
this seem to be broken on 2.x because jsonschema doe not get functools32...

**Comment by Carreau** [comment]
Add ability to build Typescript docs.

    Yeah, no, we cannot use gulp-typedocs, it has no option to exclude
    external d.ts files, so we would be docummenting jquery, CodeMirror,
    underscore... and no we cannot output the docs in `docs/` because
    typedocs complain it does not seem to be a typescript documentation
    folder.

**Comment by Carreau** [comment]
I am going to punch typescript in the face, and I am going to punch it hard.

**Comment by takluyver** [comment]
I think we're going to have to start a programmer therapy course.

**Comment by Carreau** [comment]
> I think we're going to have to start a programmer therapy course.

Yep. I'm doing Python to relax. Though once you have types, it is mentally much less exhausting to reason about the code. And it  revealed one potential bug:

```
var browser= (function() {
        if (typeof navigator === 'undefined') {
            // navigator undefined in node
            return 'None';
        }
        var N= navigator.appName, ua= navigator.userAgent, tem;
        var M= ua.match(/(opera|chrome|safari|firefox|msie)\/?\s*(\.?\d+(\.\d+)*)/i);
        if (M && (tem= ua.match(/version\/([\.\d]+)/i)) !== null) M[2]= tem[1];
        M= M? [M[1], M[2]]: [N, navigator.appVersion,'-?'];
        return M;
    })();
```

Roughly: unstable type cannot find satisfying type for return value.

**Comment by Carreau** [comment]
Ping @ellisonbg

**Comment by ellisonbg** [comment]
:)

On Tue, Jun 9, 2015 at 4:18 PM, Matthias Bussonnier <
notifications@github.com> wrote:

> Ping @ellisonbg <https://github.com/ellisonbg>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jupyter/notebook/pull/134#issuecomment-110530285>.
>



-- 
Brian E. Granger
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com

**Pull request closed without merging**